flask_oauthlib/provider/oauth2.py

```diff
@@ -408,6 +413,12 @@ def decorated(*args, **kwargs):
             except oauth2.OAuth2Error as e:
                 log.debug('OAuth2Error: %r', e)
                 return redirect(e.in_uri(redirect_uri))
+            except Exception as e:
+                log.warning('Exception caught while processing request, %s.' % e)
+                return redirect(add_params_to_uri(
+                    self.error_uri, {'error': str(e) }
+                ))
```

Comment:
@Fleurer What do you think?

Actually the `ValueError` should be fixed in OAuthLib, not in Flask-OAuthlib.